velox/common/base/SimdUtil-inl.h

```diff
@@ -1081,6 +1081,20 @@ struct ReinterpretBatch {
     return xsimd::batch<T, A>(data);
   }
 #endif
+
+#if XSIMD_WITH_NEON
+  static xsimd::batch<T, A> apply(xsimd::batch<U, A> data, const xsimd::neon&) {
+    return xsimd::batch<T, A>(data);
```

Comment:
The type for `batch<U>` and `batch<T>` are not implicitly convertible in NEON.  You need to use `vreinterpret` like we already did below.